lighthouse-core/gather/gather-runner.js

```diff
@@ -214,26 +221,55 @@ class GatherRunner {
         const artifacts = Object.assign({}, tracingData);
         passes.forEach(pass => {
           pass.gatherers.forEach(gatherer => {
+            if (typeof gatherer.artifact === 'undefined') {
```

Comment:
maybe `if ('artifact' in gatherer)` to account for gatherers that may someday want to return an undefined